### PR TITLE
Improvements for templates:

### PR DESCRIPTION
**Pull request by jdrossl**
- Update Freemarker to the latest version
- Change Freemarker configuration to prevent objects to be created
- Prevent servlet context & app context to be exposed by default
- Add a whitelist for exposing beans

craftersoftware/craftercms#71

### CODE REVIEW

File: src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Override
+    public TemplateCollectionModel values() throws TemplateModelException {
+        return new SimpleCollection(singletonList(SiteContext.getCurrent().getSiteName()), objectWrapper);
+    }
+
+    @Override
+    public TemplateModel get(String key) throws TemplateModelException {
+        if (SITE_NAME_KEY.equals(key)) {
```
Comment (avasquez614) [inline src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: Does this mean that know the only property available from `SiteContext` is `siteName`.

Comment (jdrossl) [reply src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: for freemarker, correct

Comment (avasquez614) [reply src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: @jdrossl @sumerjabri I think we should add another flag: `crafter.engine.site.expose.freemarker.siteContext`, to indicate that the `SiteContext` properties can be exposed. I'm afraid that if we ALWAYS only expose the site name in Freemarker we could be breaking backward compatibility without any way to fix it.

Comment (jdrossl) [reply src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: or we can use the same flag then

Comment (avasquez614) [reply src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: but maybe clients might want to expose other components and not the context, so 2 variables might not be bad. Anyway, I think we would be ok with one, but maybe rename it to something like `crafter.engine.restrictVariables`.

Comment (avasquez614) [inline src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: Need to check `disableVariableRestrictions` to see if all `SiteContext` properties should be exposed.

Comment (jdrossl) [reply src/main/java/org/craftercms/engine/util/freemarker/SiteContextHashModel.java:55]: The class is not used when the flag is set https://github.com/craftercms/engine/blob/acd8c10d25373f8e7077bd4a00b69734a872abf0/src/main/java/org/craftercms/engine/view/freemarker/CrafterFreeMarkerView.java#L201

**Pull request merged**